app/experimenter/experiments/models/nimbus.py

```diff
@@ -82,6 +82,31 @@ def __str__(self):
         return self.name
 
 
+class NimbusBranch(models.Model):
+    experiment = models.ForeignKey(
+        NimbusExperiment,
+        blank=False,
+        null=False,
+        related_name="branches",
+        on_delete=models.CASCADE,
+    )
+    is_control = models.BooleanField(default=False)
```

Comment:
Would it be worthwhile to put a table constraint on to ensure a given Nimbus experiment can have no more than one enabled control branch?

Reply:
Yeah I've thought about it but I don't know how to express it?  You can't use a uniqueness constraint to do it cuz you can have multiple falses?

Reply:
Unique indexes can have where statements, and maybe a check statement can do it. e.g.
```
CREATE UNIQUE INDEX only_one_true ON nimbus_branch (experiment_pk, is_control)
WHERE is_control IS true
```

Or something like that? I found this answer which discusses some approaches:
https://dba.stackexchange.com/questions/73125/restrict-two-specific-column-values-from-existing-at-the-same-time/73133#73133

Reply:
BTW, Postgres doesn't allow that format, but it does allow a unique index which is enforced the same:
https://stackoverflow.com/questions/16236365/postgresql-conditionally-unique-constraint